packages/mds-compliance-service/repository/index.ts

```diff
@@ -0,0 +1,215 @@
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      if (isDefined(compliance_snapshot_id)) {
+        const entity = await connection.getRepository(ComplianceSnapshotEntity).findOne({
+          where: {
+            compliance_snapshot_id
+          }
+        })
+        if (!entity) {
+          throw new NotFoundError(`ComplianceSnapshot ${compliance_snapshot_id} not found`)
+        }
+        return ComplianceSnapshotEntityToDomain.map(entity)
+      }
+      if (!isDefined(provider_id) || !isDefined(policy_id)) {
+        throw RepositoryError('provider_id and policy_id must be given if compliance_snapshot_id is not given')
+      }
+
+      const p_compliance_as_of = compliance_as_of || now()
+      const query = connection
+        .getRepository(ComplianceSnapshotEntity)
+        .createQueryBuilder()
+        .where(`provider_id = '${provider_id}'`)
+        .andWhere(`policy_id = '${policy_id}'`)
+        .andWhere(`compliance_as_of >= ${p_compliance_as_of}`)
+        .orderBy('compliance_as_of')
+
+      const entity = await query.getOne()
+      if (!entity) {
+        throw new NotFoundError(
+          `ComplianceSnapshot not found with params ${JSON.stringify({
+            policy_id,
+            provider_id,
+            p_compliance_as_of
+          })} not found`
+        )
+      }
+      return ComplianceSnapshotEntityToDomain.map(entity)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public updateComplianceSnapshot = async (
+    update: ComplianceSnapshotDomainModel
+  ): Promise<ComplianceSnapshotDomainModel> => {
+    const { connect } = this
+    try {
+      const connection = await connect('rw')
+      const { compliance_snapshot_id } = update
+      const currentComplianceSnapshot = await connection
+        .getRepository(ComplianceSnapshotEntity)
+        .findOne({ where: { compliance_snapshot_id } })
+      if (!currentComplianceSnapshot) {
+        throw new NotFoundError(`ComplianceSnapshot ${compliance_snapshot_id} not found`)
+      }
+      const {
+        raw: [updated]
+      } = await connection
+        .getRepository(ComplianceSnapshotEntity)
+        .createQueryBuilder()
+        .update()
+        .set({
+          ...currentComplianceSnapshot,
+          ...ComplianceSnapshotDomainToEntityCreate.map(update)
+        })
+        .where('compliance_snapshot_id = :compliance_snapshot_id', { compliance_snapshot_id })
+        .returning('*')
+        .execute()
+      return ComplianceSnapshotEntityToDomain.map(updated)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public getComplianceSnapshotsByTimeInterval = async ({
+    start_time,
+    end_time,
+    provider_ids,
+    policy_ids
+  }: GetComplianceSnapshotsByTimeIntervalOptions): Promise<ComplianceSnapshotDomainModel[]> => {
+    const { connect } = this
+    if (isDefined(start_time)) {
```

Comment:
If you made this `if(!isDefined(start_time))` instead, you could avoid nesting all of this code in the an if/else block

```typescript
if(!isDefined(start_time)) throw RepositoryError('start_time not provided')
// All your other logic without extra indentation
```